### PR TITLE
Reconnect publicize from editor

### LINKED ISSUE #6837: Publicize: Help users fix a broken publicize connection in post options
### Expected behavior

When I have a broken/non-working publicize connection, the Sharing section in the post options displays an error symbol:

I expect to be able to tap on that row to get more information about that error and fix the connection. For example, it might give me more information that the connection is broken and tell me where to go to fix it, or even take me directly to the screen in my Sharing settings where I can reconnect the service:

### Actual behavior

Tapping on that row in the sharing section highlights the row but nothing else happens. I can't find out why I get that error symbol or how to fix it. It's a dead end in the user flow.

To resolve it, I have to exit the editor and go to my site's Sharing section, open the connection, and find the option to Reconnect to that service.

### Steps to reproduce the behavior

1. In Sharing, connect to a social media service.
2. Break the connection (e.g. deactivate/delete that social media account).
3. Start a new blog post.
4. Open the post options and check the Sharing section.
5. Tap on the broken connection. Result: Nothing happens.

##### Tested on iPhone 6, iOS 10.2.1, WPiOS 7.1.0.20170306

### PR DESCRIPTION
**Pull request by koke**
We already show if a connection is broken from post options. This lets the user fix the issue right there. I had to refactor things a little bit as the sharing detail expected the publicize services to be synced, so I removed that requirement. It's still not bulletproof as the sync can fail and there's no retry, but I'd leave that for the new network infrastructure.

I also had to fix this crazy hard to debug thing where the same authorization web view would work from the Sharing section but not when presented from the editor, because the editor enables PrivateSiteURLProtocol. I updated it so it forwards redirects to the client, but perhaps we should limit it to wordpress.com sites (i.e. exclude public-api).

Fixes #6837 

To test:

- Do a fresh install to ensure publicize services aren't cached
- Break one of the publicize connections on the test account (e.g. remove WordPress from https://www.facebook.com/settings?tab=applications)
- Launch the app and log in
- Create a new post
- Go to post options
- Notice the exclamation icon next to the broken account and tap on it
- The sharing details view shows
- Tap on Reconnect
- A web browser pops up to reconnect the account

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/SharingDetailViewController.m [hunk 1]
```diff
@@ -180,7 +174,37 @@ - (void)reconnectPublicizeConnection
...
+                                             success:attemptReconnect
+                                             failure:^(NSError * _Nullable error) {
+                                                 [WPError showNetworkingAlertWithError:error];
+                                             }];
+    } else {
+        attemptReconnect();
+    }
+}
+
+- (SharingAuthorizationHelper *)authorizationHelper
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Blog/SharingDetailViewController.m]: The only reason we keep the property is so we can set the delegate to nil on dealloc.

Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SharingDetailViewController.m]: I wonder if this is necessary since the delegate is already a weak reference.

**Comment by koke** [comment]
OK, here's another go at it, without the property. This felt like trying to write Swift optional style code in Obj-C, and it's so weird. Also, I'm so ready to move past success/failure async programming 😬

**Comment by aerych** [comment]
:shipit: yo :)

**Pull request merged**